File: works/tmp/1773.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using static System.Console;

namespace ConsoleApp
{
    class Program
    {

        static void Main(string[] args)
        {
            bool[] mark = new bool[2000010];
            bool[] students = new bool[2000010];
            string[] _s = ReadLine().Split(' ');
            int n = int.Parse(_s[0]);
            int c = int.Parse(_s[1]);
            int cnt = 0;

            for (int i = 0; i < n; i++)
            {
                int num = int.Parse(ReadLine());
                if (students[num]) continue;
                students[num] = true;
                for (int j = num; j <= c; j += num)
                {
                    if (!mark[j])
                    {
                        cnt++;
                        mark[j] = true;
                    }
                }
            }
            Write(cnt);
        }

    }
}

File: works/tmp/1773.py
import sys

mark = [False for _ in range(2000010)]
students = [False for _ in range(2000010)]
n, c = map(int, input().split())
cnt = 0

for i in range(0, n):
    num = int(sys.stdin.readline())
    if students[num]: continue
    students[num] = True
    for j in range(num, c+1, num):
        if not mark[j]:
            cnt += 1
            mark[j] = True

print(cnt)